tests/e2e/c/proposervm_epoch.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package c
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ava-labs/coreth/ethclient"
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/api/info"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/proposervm"
+)
+
+var _ = e2e.DescribeCChain("[ProposerVM Epoch]", func() {
+	tc := e2e.NewTestContext()
+	require := require.New(tc)
+
+	const txAmount = 10 * units.Avax // Arbitrary amount to send and transfer
+
+	ginkgo.It("should advance the proposervm epoch according to the upgrade config epoch duration", func() {
+		env := e2e.GetEnv(tc)
+		senderKey := env.PreFundedKey
+		recipientKey := e2e.NewPrivateKey(tc)
+		nodeURI := env.GetRandomNodeURI()
+		ethClient := e2e.NewEthClient(tc, nodeURI)
+
+		infoClient := info.NewClient(nodeURI.URI)
+		upgrades, err := infoClient.Upgrades(tc.DefaultContext())
+		require.NoError(err)
+
+		// Don't run test if granite isn't active.
+		if !upgrades.IsGraniteActivated(time.Now()) {
+			return
+		}
+
+		// Genesis does not contain a proposervm block
+		for blockNumber(tc, ethClient) == 0 {
```

Comment:
This has the potential to loop endlessly, so perhaps we should use an upper limit to how much we can loop?